lib/bamboo/adapters/sendgrid_adapter.ex

```diff
@@ -89,6 +56,34 @@ defmodule Bamboo.SendgridAdapter do
     end
   end
 
+  defp raise_api_error(message), do: raise(ApiError, message: message)
+  defp raise_api_error(params, response) do
+    filtered_params = params |> Plug.Conn.Query.decode |> Map.put("key", "[FILTERED]")
+
+    message = """
+    There was a problem sending the email through the SendGrid API.
```

Comment:
What do you think about extracting all of these info the `ApiError` itself? Passing in the params (pre-filtered or parsed), response, name of the adapter and some additional information would remove some of this duplication.

I think it also provides a nice template that other adapters can follow without any additional work by just passing in the params and response.